client/my-sites/post-type-list/index.jsx

```diff
@@ -8,255 +8,214 @@ import PropTypes from 'prop-types';
+			! isEqual( this.props.query, nextProps.query ) ||
+			! isEqual( this.props.siteId, nextProps.siteId )
+		) {
+			const maxRequestedPage = this.estimatePageCountFromPosts(
+				nextProps.posts
+			);
+			this.setState( {
+				maxRequestedPage,
+			} );
+		}
 	}
 
-	componentDidMount() {
-		if ( this.props.wrapTitles ) {
-			// Note: Assumes that this property does not change
-			this.resizeListener = throttle( this.handleWindowResize, 50 );
-			window.addEventListener( 'resize', this.resizeListener );
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.isRequestingPosts && ! this.props.isRequestingPosts ) {
+			// We just finished loading a page.  If the bottom of the list is
+			// still visible on screen (or almost visible), then we should go
+			// ahead and load the next page.
+			this.maybeLoadNextPage();
 		}
 	}
 
-	componentWillReceiveProps( nextProps ) {
-		if ( ! isEqual( this.props.query, nextProps.query ) ) {
-			this.setState( {
-				requestedPages: this.getInitialRequestedPages( nextProps ),
-			} );
-		}
+	componentDidMount() {
+		this.maybeLoadNextPage();
 	}
 
 	componentWillUnmount() {
-		if ( this.resizeListener ) {
-			window.removeEventListener( 'resize', this.resizeListener );
-			delete this.resizeListener;
-		}
+		window.removeEventListener( 'scroll', this.scrollListener );
+		this.scrollListener.cancel(); // Cancel any pending scroll events
 	}
 
-	handleWindowResize = () => {
-		this.setState( {
-			windowWidth: window.innerWidth,
-		} );
-	};
+	estimatePageCountFromPosts( posts ) {
+		// When loading posts from persistent storage, we want to avoid making
+		// a bunch of sequential requests when the user scrolls down to the end
+		// of the list.  However, we want to still request the posts, in case
+		// some data has changed since the last page reload.  This will spawn a
+		// number of concurrent requests for different pages of the posts list.
 
-	getInitialRequestedPages( props ) {
-		// If we have no posts or we're otherwise not expecting any posts to be
-		// rendered, request the first page, since setRequestedPages won't be
-		// called if row count is 0.
-		if ( 0 === size( props.posts ) ) {
-			return [ 1 ];
+		if ( ! posts || ! posts.length ) {
+			return 1;
 		}
 
-		return [];
-	}
+		const query = this.props.query || {};
+		const postsPerPage = query.number || DEFAULT_POST_QUERY.number;
+		const pageCount = Math.ceil( posts.length / postsPerPage );
 
-	getPageForIndex( index ) {
-		const { query, lastPage } = this.props;
-		const perPage = query.number || DEFAULT_POSTS_PER_PAGE;
-		const page = Math.ceil( index / perPage );
+		// Avoid making more than 5 concurrent requests on page load.
+		return Math.min( pageCount, 5 );
+	}
 
-		return Math.max( Math.min( page, lastPage || Infinity ), 1 );
+	getScrollTop() {
+		const { scrollContainer } = this.props;
+		if ( ! scrollContainer ) {
+			return null;
+		}
+		if ( scrollContainer === document.body ) {
+			return 'scrollY' in window
+				? window.scrollY
```

Comment:
Happy to see this fixed the issues in latest Chrome 🎉

Reply:
Thanks for re-testing!